activesupport/lib/active_support/broadcast_logger.rb

```diff
@@ -218,6 +218,17 @@ def fatal!
       dispatch { |logger| logger.fatal! }
     end
 
+    def deep_dup
+      duplicate = dup
```

Comment:
How about just doing `self.class.new(@broadcasts.map(&:dup)`?
To me, `BroadcastLogger` is a rather lightweight shell around the loggers, so there's not much value in duplicating it. 
This way you'd avoid having to call `#stop_broadcasting_to`. (Also note that `#stop_broadcasting_to` is equivalent to removing an item from `@broadcasts`.

Reply:
Ah, I just realized that calling `dup` like you do is not going to work. Even `@broadcasts` is not duplicated:
```rb
bl1 = ActiveSupport::BroadcastLogger.new
bl2 = bl1.dup

bl1.broadcasts.__id__ == bl2.broadcasts.__id__
#=> true
```